stats/albatross_stats_stubs.c

```diff
@@ -150,23 +150,47 @@ CAMLprim value vmmanage_vmmapi_stats (value octx) {
   CAMLreturn(res);
 }
 
-CAMLprim value vmmanage_sysctl_ifcount (value unit) {
-  CAMLparam1(unit);
-  int data = 0;
+CAMLprim value vmmanage_get_ifindex_by_name (value name) {
+  CAMLparam1(name);
+  const char *devname;
+  int ifcount = 0;
   size_t dlen = 0;
-  int name[5];
+  int name_ifcount[5];
+  int name_ifdata[6];
+  struct ifmibdata data;
 
-  name[0] = CTL_NET;
-  name[1] = PF_LINK;
-  name[2] = NETLINK_GENERIC;
-  name[3] = IFMIB_SYSTEM;
-  name[4] = IFMIB_IFCOUNT;
-  dlen = sizeof(data);
+  if (! caml_string_is_c_safe(name)) caml_raise_not_found();
 
-  if (sysctl(name, nitems(name), &data, &dlen, NULL, 0) != 0)
+  devname = String_val(name);
+
+  name_ifcount[0] = CTL_NET;
+  name_ifcount[1] = PF_LINK;
+  name_ifcount[2] = NETLINK_GENERIC;
+  name_ifcount[3] = IFMIB_SYSTEM;
+  name_ifcount[4] = IFMIB_IFCOUNT;
+  dlen = sizeof(ifcount);
+
+  if (sysctl(name, nitems(name_ifcount), &ifcount, &dlen, NULL, 0) != 0)
     uerror("sysctl", Nothing);
 
-  CAMLreturn(Val_long(data));
+  for (int idx = 1; idx <= ifcount; idx++) {
```

Comment:
maybe smarter to go from last index (ifcount) to 1 (or 0?)? The intuition is that this code is called for newly created virtual machines, that use fresh tap devices, and I expect the tap device numbering to be linear and monotonically increasing.